src/core/lib/event_engine/thread_pool/work_stealing_thread_pool.cc

```diff
@@ -125,6 +168,11 @@ thread_local WorkQueue* g_local_queue = nullptr;
 
 WorkStealingThreadPool::WorkStealingThreadPool(size_t reserve_threads)
     : pool_{std::make_shared<WorkStealingThreadPoolImpl>(reserve_threads)} {
+  static bool logged = false;
```

Comment:
is this class's ctor (i.e. `MakeThreadPool`) not thread safe? (wondering if `logged` should be atomic)

Reply:
Good catch. I removed the check, and now only trace-log the message ... on every thread pool instantiation. There's usually one, except for in tests. I don't think we have a good pattern today for a thread-safe post-grpc-init once-init situation, which is what I'd reach for.

Reply:
> I don't think we have a good pattern today for a thread-safe post-grpc-init once-init situation, which is what I'd reach for.

Maybe `gpr_once` ?

Current code LGTM though